server/notification-providers/flashduty.js

```diff
@@ -0,0 +1,92 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+const { UP, DOWN, getMonitorRelativeURL } = require("../../src/util");
+const { setting } = require("../util-server");
+const successMessage = "Sent Successfully.";
+
+class FlashDuty extends NotificationProvider {
+    name = "FlashDuty";
+
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        try {
+            if (heartbeatJSON == null) {
+                const title = "Uptime Kuma Alert";
+                const monitor = {
+                    type: "ping",
+                    url: msg,
+                };
+                return this.postNotification(notification, title, msg, monitor);
+            }
+
+            if (heartbeatJSON.status === UP) {
+                const title = "Uptime Kuma Monitor ✅ Up";
+
+                return this.postNotification(notification, title, heartbeatJSON.msg, monitorJSON, "Ok");
+            }
+
+            if (heartbeatJSON.status === DOWN) {
+                const title = "Uptime Kuma Monitor 🔴 Down";
+                return this.postNotification(notification, title, heartbeatJSON.msg, monitorJSON, notification.flashdutySeverity);
+            }
+        } catch (error) {
+            this.throwGeneralAxiosError(error);
+        }
+    }
+
+    genMonitorUrl(monitorInfo) {
```

Comment:
I am unsure if the undefined case works correctly with the notification provider
See https://github.com/louislam/uptime-kuma/issues/3274 for an instance of what is my concern ^^
(also unsure if url were null or undefined in this case…)

```suggestion
    /**
     * generate a monitor url from the monitors infomation
     * @param {Object} monitorInfo Monitor details
     * @returns {string|undefined} url linking to the monitor
     */
    genMonitorUrl(monitorInfo) {
```

Reply:
When Testing, the url is undefined, but when the monitor up or down ,the url will be got successfully. as the below screenshot. the param description has been appended.

![](https://i.imgur.com/dWKUbdk.png)